cmd/juju/commands/deploy_test.go

```diff
@@ -499,9 +500,21 @@ func (s *charmStoreSuite) SetUpTest(c *gc.C) {
 	})
 
 	// Set up the charm store testing server.
-	s.srv = charmstoretesting.OpenServer(c, s.Session, charmstore.ServerParams{
+	db := s.Session.DB("juju-testing")
+	params := charmstore.ServerParams{
+		AuthUsername:     "test-user",
+		AuthPassword:     "test-password",
 		IdentityLocation: s.discharger.Location(),
 		PublicKeyLocator: s.discharger,
+	}
+	handler, err := charmstore.NewServer(db, nil, "", params, charmstore.V4)
```

Comment:
It's V4. It's not the library version, it's the HTTP API version. I know, it's confusing.